Error build with Tesseract

Windows 10 x64 LTSB, VS 2017 15.9.3, CMake 3.12.0, opencv rev dad2247b562c977a13cf940a17a19880914c648f. Tesseract ver 4.1 and Leptonica 1.79. 

Opencv was configured via Cmake gui and builded through msbuild.
CMake gui changes:
````-DOPENCV_EXTRA_MODULES_PATH:PATH="E:/Lib_prebuild/opencv/source/opencv_contrib/modules" -DBUILD_EXAMPLES:BOOL="1" -DLept_LIBRARY:FILEPATH="optimized;E:/Lib_prebuild/leptonica/build_x64/lib/leptonica-1.79.0.lib;debug;E:/Lib_prebuild/leptonica/build_x64/lib/leptonica-1.79.0d.lib" -DINSTALL_PYTHON_EXAMPLES:BOOL="1" -DTesseract_INCLUDE_DIR:PATH="E:/Lib_prebuild/tesseract/build_x64/include" -DTesseract_LIBRARY:FILEPATH="optimized;E:/Lib_prebuild/tesseract/build_x64/lib/tesseract41.lib;debug;E:/Lib_prebuild/tesseract/build_x64/lib/tesseract41d.lib" -DINSTALL_C_EXAMPLES:BOOL="1" -DBUILD_TESTS:BOOL="0" -DBUILD_PERF_TESTS:BOOL="0" ````


<details>
  <summary>build errors:</summary>

````
"E:\000ocv\prebuild_x64\OpenCV.sln" (default target) (1) ->
       "E:\000ocv\prebuild_x64\ALL_BUILD.vcxproj.metaproj" (default target) (2) ->
       "E:\000ocv\prebuild_x64\samples\cpp\example_cpp_3calibration.vcxproj.metaproj" (default target) (69) ->
       "E:\000ocv\prebuild_x64\modules\datasets\opencv_datasets.vcxproj.metaproj" (default target) (70) ->
       "E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj.metaproj" (default target) (79) ->
       "E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj" (default target) (556) ->
       (ClCompile target) ->
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(41): error C2589: '(': ille
       gal token on right side of '::' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(41): error C2062: type 'unk
       nown-type' unexpected [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(41): error C2059: syntax er
       ror: ')' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(42): error C2589: '(': ille
       gal token on right side of '::' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(42): error C2062: type 'unk
       nown-type' unexpected [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(42): error C2059: syntax er
       ror: ')' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(44): error C2589: '(': ille
       gal token on right side of '::' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(44): error C2062: type 'unk
       nown-type' unexpected [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(44): error C2059: syntax er
       ror: ')' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(45): error C2589: '(': ille
       gal token on right side of '::' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(45): error C2062: type 'unk
       nown-type' unexpected [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\text_detectorcnn.cpp(45): error C2059: syntax er
       ror: ')' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\erfilter.cpp(1412): error C2589: '(': illegal to
       ken on right side of '::' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\erfilter.cpp(1412): error C2062: type 'unknown-t
       ype' unexpected [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]
         e:\lib_prebuild\opencv\source\opencv_contrib\modules\text\src\erfilter.cpp(1412): error C2059: syntax error: '
       )' [E:\000ocv\prebuild_x64\modules\text\opencv_text.vcxproj]

    2 Warning(s)
    15 Error(s)

````
</details>

Look like macro min and max defined in 3rd party lib (Leptonica, Tesseract, libpng...)
Adding
#undef min and #undef max
resolve this, but is it a good idea to use them in Opencv Text module?
Create the appropriate pull request?

--- a/modules/text/src/ocr_tesseract.cpp
+++ b/modules/text/src/ocr_tesseract.cpp
@@ -48,6 +48,14 @@
 #include <fstream>
 #include <queue>
 
+#ifdef HAVE_TESSERACT
+#if !defined(USE_STD_NAMESPACE)
+#define USE_STD_NAMESPACE
+#endif
+#include <tesseract/baseapi.h>
+#include <tesseract/resultiterator.h>
+#endif
+
 namespace cv
 {
 namespace text

--- a/modules/text/src/precomp.hpp
+++ b/modules/text/src/precomp.hpp
@@ -47,12 +47,4 @@
 
 #include "text_config.hpp"
 
-#ifdef HAVE_TESSERACT
-#if !defined(USE_STD_NAMESPACE)
-#define USE_STD_NAMESPACE
-#endif
-#include <tesseract/baseapi.h>
-#include <tesseract/resultiterator.h>
-#endif
-
 #endif

